src/Microsoft.Health.Fhir.Azure/ExportDestinationClient/AzureExportDestinationClient.cs

```diff
@@ -66,193 +63,77 @@ public async Task ConnectAsync(ExportJobConfiguration exportJobConfiguration, Ca
         {
-            CheckIfClientIsConnected();
-
-            try
-            {
-                await Policy.Handle<StorageException>()
-                    .WaitAndRetryAsync(
-                        retryCount: 2,
-                        sleepDurationProvider: (retryCount) => TimeSpan.FromSeconds(5 * (retryCount - 1)),
-                        onRetryAsync: async (exception, retryCount) =>
-                        {
-                            _logger.LogWarning(exception, "Error while uploading blobs.");
-                            await RefreshClientAsync(exportJobConfiguration, cancellationToken);
-                        })
-                    .ExecuteAsync(() => UploadBlobHelperAsync(cancellationToken));
-            }
-            catch (DestinationConnectionException)
-            {
-                throw;
-            }
-            catch (Exception ex)
+            if (_dataBuffers.ContainsKey(fileName))
             {
-                _logger.LogError(ex, "Can't get an authorized client");
-                throw new DestinationConnectionException(Resources.CannotGetAuthorizedClient, HttpStatusCode.Unauthorized);
-            }
-
-            try
-            {
-                await Policy.Handle<StorageException>()
-                    .WaitAndRetryAsync(
-                        retryCount: 2,
-                        sleepDurationProvider: (retryCount) => TimeSpan.FromSeconds(5 * (retryCount - 1)),
-                        onRetryAsync: async (exception, retryCount) =>
-                        {
-                            _logger.LogWarning(exception, "Error while committing block list.");
-                            await RefreshClientAsync(exportJobConfiguration, cancellationToken);
-                        })
-                    .ExecuteAsync(() => UploadBlockListHelperAsync(cancellationToken));
-            }
-            catch (DestinationConnectionException)
-            {
-                throw;
-            }
-            catch (Exception ex)
-            {
-                _logger.LogError(ex, "Can't get an authorized client");
-                throw new DestinationConnectionException(Resources.CannotGetAuthorizedClient, HttpStatusCode.Unauthorized);
-            }
-
-            // If we are here, we managed to upload and commit everything successfully.
-            // We can clear the stream mappings once we commit everything.
-            foreach (Stream stream in _streamMappings.Values)
-            {
-                stream.Dispose();
-            }
+                BlockBlobClient blockBlob = _blobContainer.GetBlockBlobClient(fileName);
 
-            _streamMappings.Clear();
-        }
-
-        private async Task UploadBlobHelperAsync(CancellationToken cancellationToken)
-        {
-            // Upload all blocks for each blob that was modified.
-            Task[] uploadTasks = new Task[_streamMappings.Count];
+                using var stream = blockBlob.OpenWrite(true);
```

Comment:
Since we need the stream right away there isn't any benefit to using OpenWriteAsync. If we use the OpenWriteAsync we would have to await it, which would make it synchronous.